lib/modules/manager/unity3d/extract.spec.ts

```diff
@@ -0,0 +1,159 @@
+        beta,
+        'ProjectSettings/ProjectVersion.txt',
+      )?.deps;
+      expect(res).toEqual([
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: '2023.3.0b5',
+          datasource: 'unity3d',
+          depName: 'm_EditorVersion',
+          depType: 'final',
+          replaceString: 'm_EditorVersion: 2023.3.0b5',
+        },
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: '2023.3.0b5 (30acc77e9b6b)',
+          datasource: 'unity3d',
+          depName: 'm_EditorVersionWithRevision',
+          depType: 'final',
+          replaceString:
+            'm_EditorVersionWithRevision: 2023.3.0b5 (30acc77e9b6b)',
+        },
+      ]);
+    });
+
+    it('handles stable versions', () => {
+      const res = extractPackageFile(
+        stable,
+        'ProjectSettings/ProjectVersion.txt',
+      )?.deps;
+      expect(res).toEqual([
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: '2022.3.19f1',
+          datasource: 'unity3d',
+          depName: 'm_EditorVersion',
+          depType: 'final',
+          replaceString: 'm_EditorVersion: 2022.3.19f1',
+        },
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: '2022.3.19f1 (244b723c30a6)',
+          datasource: 'unity3d',
+          depName: 'm_EditorVersionWithRevision',
+          depType: 'final',
+          replaceString:
+            'm_EditorVersionWithRevision: 2022.3.19f1 (244b723c30a6)',
+        },
+      ]);
+    });
+
+    it('handles lts versions', () => {
+      const res = extractPackageFile(
+        lts,
+        'ProjectSettings/ProjectVersion.txt',
+      )?.deps;
+      expect(res).toEqual([
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: '2021.3.35f1',
+          datasource: 'unity3d',
+          depName: 'm_EditorVersion',
+          depType: 'final',
+          replaceString: 'm_EditorVersion: 2021.3.35f1',
+        },
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: '2021.3.35f1 (157b46ce122a)',
+          datasource: 'unity3d',
+          depName: 'm_EditorVersionWithRevision',
+          depType: 'final',
+          replaceString:
+            'm_EditorVersionWithRevision: 2021.3.35f1 (157b46ce122a)',
+        },
+      ]);
+    });
+
+    it('skips if casing is incorrect', () => {
```

Comment:
Should Renovate raise a warning about the malformed casing? Or should Unity warn about malformed files?